src/hooks/use-click-outside/__tests__/index.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { useRef } from 'react'
+import { useClickOutside } from '..'
+import { fireEvent, render } from '@testing-library/react'
+
+const TestComponent = ({ onClickOutside, isUsingParent = true }) => {
+  const ref = useRef<HTMLDivElement>(null)
+  useClickOutside(ref, onClickOutside)
+
```

Comment:
**note:** When writing tests, it's generally good practice to avoid our test code having conditionals in it, as it makes it harder for us to know whether our subject-under-test is behaving correctly, or if our conditional test code is getting in the way.

**suggestion:** I think it would be better to have more than one test component instead, or to do away with the `isUsingParent` prop and just have the same DOM content rendered no matter what (see my comments below for why this may be an option)